### PR TITLE
Fix #16327: Crash supplying a bad signature size in the auth packet

### LINKED ISSUE #16327: [bcdb78] openrct2.exe: NetworkBase::Server_Handle_AUTH, NetworkBase::ProcessPacket, NetworkBase::ProcessConnection, NetworkBa...
<table>
    <tr>
        <td></td>
        <td><b><a href="https://openrct2.sp.backtrace.io/dashboard/openrct2/project/openrct2/group/bcdb78bc98da8f04643da0bc0377e1a41ef8d922ed6f0bffb85902871258a4fd">New error: [openrct2]</a></b><br /><b>1</b> occurrence since (Fri Nov 15 2019 11:27:39)<br /></td>
    </tr>
</table><table>
    <tr><td><strong>Application</strong></td><td>openrct2.exe</td></tr>
<tr>
    <td><strong>Attributes</strong></td>
    <td>
<strong>attachment_config.ini</strong> &mdash; 1
        <ul>
                <li><code>C:\Users\remco\Documents\OpenRCT2\crash\b9ba90a6-bbcf-49ab-a7a6-45eb32480948.ini</code> &mdash; 1</li></ul><br>
<strong>assert_failure</strong> &mdash; 1
        <ul>
                <li><code>*</code> &mdash; 1</li></ul><br>
<strong>fault.address</strong> &mdash; 1
        <ul>
                <li><code>2</code> &mdash; 1</li></ul><br>
<strong>version</strong> &mdash; 1
        <ul>
                <li><code>OpenRCT2, v0.3.5.1 (61c67af) provided by GitHub</code> &mdash; 1</li></ul><br>
<strong>error.message</strong> &mdash; 1
        <ul>
                <li><code>EXCEPTION_ACCESS_VIOLATION_READ</code> &mdash; 1</li></ul><br>
<strong>commit</strong> &mdash; 1
        <ul>
                <li><code>61c67af</code> &mdash; 1</li></ul><br></td>
    </tr>
    <tr>
        <td><strong>Classifiers</strong></td>
        <td>
            <strong>invalid-read</strong><br></td>
    </tr>
    <tr>
        <td><strong>Frames</strong></td>
        <td>
            <code>
NetworkBase::Server_Handle_AUTH
NetworkBase::ProcessPacket
NetworkBase::ProcessConnection
NetworkBase::UpdateServer
network_update
OpenRCT2::GameState::Update
OpenRCT2::Context::Update
OpenRCT2::Context::RunGameLoop
OpenRCT2::Context::RunOpenRCT2
NormalisedMain
wmain
TerminateProcessOnMemoryExhaustion</code>
        </td>
    </tr></table>

### PR DESCRIPTION
**Pull request by ZehMatt**
No network version bump required

Closes #16327

### CODE REVIEW

File: src/openrct2/network/NetworkBase.cpp [hunk 1]
```diff
@@ -2559,6 +2559,15 @@ void NetworkBase::Server_Handle_AUTH(NetworkConnection& connection, NetworkPacke
         {
             try
             {
+                // RSA technically supports keys up to 65536 bits, so this the
+                // maximum signature size for now.
```
Comment (Broxzier) [inline src/openrct2/network/NetworkBase.cpp:2563]:
```suggestion
                // RSA technically supports keys up to 65536 bits, so this is the
                // maximum signature size for now.
```

**Review by Broxzier — COMMENTED** [review]
Change looks sane, only noticed a small typo.

**Review by janisozaur — COMMENTED** [review]
Why do we need this? Can't we support the larger keys?

**Comment by ZehMatt** [comment]
Because one can send malformed packets where the signature size is something like 0xFFFFFFFF forcing to either fail to allocate or actually allocate 4 GiB memory.

**Review by Broxzier — APPROVED** [review]

**Pull request merged**